nixos/modules/services/wayland/cage.nix

```diff
@@ -0,0 +1,82 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cage;
+in {
+  options.services.cage.enable = mkEnableOption ''
+    Enable the Cage Kiosk service.
+  '';
+
+  options.services.cage.user = mkOption {
+    type = types.str;
+    default = "demo";
+    description = ''
+      User to log-in as.
+    '';
+  };
+
+  options.services.cage.program = mkOption {
+    type = types.path;
+    default = "${pkgs.xterm}/bin/xterm";
```

Comment:
It's a bit of an odd default to start an X11 application - don't we have a lightweight wayland terminal?

Reply:
I kept it that way in #81327.